lua/nui/popup/popup.md

```diff
@@ -0,0 +1,460 @@
+```
+
+- Relative to the window of `id`
+
+```lua
+relative = {
+  type = "win",
+  winid = 5,
+}
+```
+
+- Relative to the buffer's `position`
+
+```lua
+relative = {
+  type = "buf",
+  position = {
+    row = 5,
+    col = 5,
+  },
+}
+```
+
+---
+
+#### `position`
+
+- **Type:** `number` or `percentage string` or `table`
+- If `position` is `number` or `percentage string`, it applies to both `row` and `col`
+- Position is calculated from the top-left corner
+- For `percentage`, position is calculated according to the option `relative`
+  - If `relative` is set to `"buf"` or `"cursor"`, `percentage string` is not allowed
+
+**Examples**
+
+```lua
+position = 50,
+```
+
+```lua
+position = "50%",
+```
+
+```lua
+position = {
+  row = 30,
+  col = 20
+}
+```
+
+```lua
+position = {
+  row = "30%",
+  col = "20%"
+}
+```
+
+---
+
+### `size`
+
+- **Type:** `number` or `percentage string` or `table`
+- Determines the size of the popup
+- For `percentage`, `size` is calculated according to the option `relative`
+  - If `relative` is set to `buf` or `cursor`, window `size` is considered
+- If `size` is `number` or `percentage`, it applies to both `width` and `height`
+  - You can also pass a table to set them separately.
+
+**Examples**
+
+```lua
+size = 50,
+```
+
+```lua
+size = "50%",
+```
+
+```lua
+size = {
+  width = 30,
+  height = 50,
+}
+```
+
+```lua
+size = {
+  width = "30%",
+  height = "50%",
+}
+```
+
+---
+
+### `enter`
+
+- **Type:** `boolean`
+- If `true`, the popup is auto-entered after mount
+
+**Examples**
+
+```lua
+enter = true,
+```
+
+---
+
+### `focusable`
+
+- **Type:** `boolean`
+- Is `false`, the popup can not be entered by user actions (wincmds, mouse events)
+
+**Examples**
+
+```lua
+focusable = true,
+```
+
+---
+
+### `zindex`
+
+- **Type:** `number`
+- Sets the order of the popup on z-axis
+- Popup with higher the `zindex` goes on top of popups with lower `zindex`
+
+**Examples**
+
+```lua
+zindex = 20,
+```
+
+---
+
+### `buf_options`
+
+- **Type:** `table`
+- Contains all buffer related options from native buffer options `:h options /local to buffer`
+
+**Examples**
+
+```lua
+buf_options = {
+  modifiable = false,
+  readonly = true,
+},
+```
+
+---
+
+### `win_options`
+
+- **Type:** `table`
+- Contains all window related options from native buffer options `:h options /local to window`
+
+**Examples**
+
+```lua
+win_options = {
+  winblend = 10,
+  winhighlight = "Normal:Normal",
+},
+```
+
+## Method Descriptions
+
+### `popup:mount()`
+
+- Mounts the popup
+
+```lua
+popup:mount()
+```
+
+---
+
+### `popup:unmount()`
+
+- Unmounts the popup
+
+```lua
+popup:unmount()
+```
+
+---
+
+### `popup:map(mode, key, handler, opts, force)`
+
+- Sets keymap for this popup.
+- If keymap was already set and `force` is not `true` it returns `false`, otherwise returns `true`
+
+```lua
+local ok = popup:map("n", "<esc>", function(bufnr)
+  print("ESC pressed in Normal mode!")
+end, { noremap = true })
+```
+
+#### Options
```

Comment:
These can be just bold texts instead of header. These headers look like this in Table of Content:

![image](https://user-images.githubusercontent.com/8050659/130230200-7907da2c-c894-4049-9e93-3220194d033e.png)